integration/simulation/transaction_injector.go

```diff
@@ -296,18 +298,30 @@ func (m *TransactionInjector) createInvalidSignage(tx types.TxData, w wallet.Wal
 }
 
 func (m *TransactionInjector) rndObsWallet() wallet.Wallet {
+	if len(m.obsWallets) == 1 {
```

Comment:
`rand.Intn` spews if the len is zero, here and below.

Reply:
I think the `rand.Intn()` usage isn't quite right here, docs says it gives an int in the half-open interval `[0, n)` so it's safe to use it like `return m.obsWallets[rand.Intn(len(m.obsWallets))]` and with the current way we'd never get the last element in the slice I think. That would let you get rid of the length 1 handling.